apps/developer-hub/src/components/EntropyTable/index.tsx

```diff
@@ -0,0 +1,125 @@
+"use client";
+import { InfoBox } from "@pythnetwork/component-library/InfoBox";
+import { Spinner } from "@pythnetwork/component-library/Spinner";
+import type {
+  ColumnConfig,
+  RowConfig,
+} from "@pythnetwork/component-library/Table";
+import { Table } from "@pythnetwork/component-library/Table";
+import { useEffect, useRef, useState } from "react";
+
+import { FORTUNA_API_URLS } from "./constants";
+import type { EntropyDeployment } from "./entropy-api-data-fetcher";
+import { fetchEntropyDeployments } from "./entropy-api-data-fetcher";
+import CopyAddress from "../CopyAddress";
+
+export const EntropyTable = ({ isMainnet }: { isMainnet: boolean }) => {
+  const isLoading = useRef(false);
+  const [state, setState] = useState<State>(State.NotLoaded());
+
+  useEffect(() => {
+    if (!isLoading.current) {
+      setState(State.Loading());
+      isLoading.current = true;
+      getEntropyDeployments(isMainnet)
+        .then((chains) => {
+          setState(State.Loaded(chains));
+        })
+        .catch((error: unknown) => {
+          setState(State.Failed(error));
+        });
+    }
+  }, [isMainnet]);
+
+  switch (state.type) {
+    case StateType.Loading:
+    case StateType.NotLoaded: {
+      return <Spinner label="Fetching the list of entropy contracts..." />;
+    }
+    case StateType.Error: {
+      return (
+        <InfoBox title="Error" variant="error">
+          <p>Failed to fetch the list of entropy contracts.</p>
+        </InfoBox>
+      );
+    }
+    case StateType.Loaded: {
+      return <EntropyTableContent chains={state.chains} />;
+    }
+    default: {
+      throw new Error(`Unknown state type: ${typeof state}`);
+    }
```

Comment:
This is not necessary and you should not have it as it will break typescript's ability to do exhaustivity checks (e.g. if you add a state, you want typescript to tell you that you aren't handling it.  If you have a default case, typescript doesn't know you aren't handling it and thinks you're handling all cases, and it won't tell you)